Taskfile.yml

```diff
@@ -84,3 +84,92 @@ tasks:
+          $(
+            # The odd method for escaping . in the regex is required for windows compatibility because mvdan.cc/sh gives
+            # \ characters special treatment on Windows in an attempt to support them as path separators.
+            find . \
+              -type d -name '.git' -prune -or \
+              -type d -name '.licenses' -prune -or \
+              -type d -name '__pycache__' -prune -or \
+              -type d -name 'node_modules' -prune -or \
+              -type d -path './deps/*' -prune -or \
+              \( \
+                -regextype posix-extended \
+                -regex '.*[.](bash|sh)' -and \
+                -type f \
+              \) \
+              -print
+          )
+
+  # Source: https://github.com/arduino/tooling-project-assets/blob/main/workflow-templates/assets/check-shell-task/Taskfile.yml
+  shell:check-mode:
+    desc: Check for non-executable shell scripts
+    cmds:
+      - |
+        EXIT_STATUS=0
+        while read -r nonExecutableScriptPath; do
+          # The while loop always runs once, even if no file was found
+          if [[ "$nonExecutableScriptPath" == "" ]]; then
+            continue
+          fi
+
+          echo "::error file=${nonExecutableScriptPath}::non-executable script file: $nonExecutableScriptPath";
+          EXIT_STATUS=1
+        done <<<"$(
+          # The odd approach to escaping `.` in the regex is required for windows compatibility because mvdan.cc/sh
+          # gives `\` characters special treatment on Windows in an attempt to support them as path separators.
+          find . \
+            -type d -name '.git' -prune -or \
+            -type d -name '.licenses' -prune -or \
+            -type d -name '__pycache__' -prune -or \
+            -type d -name 'node_modules' -prune -or \
+            -type d -path './deps/*' -prune -or \
+            \( \
+              -regextype posix-extended \
+              -regex '.*[.](bash|sh)' -and \
+              -type f -and \
+              -not -executable \
+              -print \
+            \)
+        )"
+        exit $EXIT_STATUS
+
+  # Source: https://github.com/arduino/tooling-project-assets/blob/main/workflow-templates/assets/check-shell-task/Taskfile.yml
+  shell:format:
+    desc: Format shell script files
+    cmds:
+      - |
+        if ! which shfmt &>/dev/null; then
+          echo "shfmt not installed or not in PATH. Please install: https://github.com/mvdan/sh#shfmt"
+          exit 1
+        fi
+      - |
+        find . \
+          -type d -name '.git' -prune -or \
+          -type d -name '.licenses' -prune -or \
+          -type d -name '__pycache__' -prune -or \
+          -type d -name 'node_modules' -prune -or \
+          -type d -path './deps/*' -prune -or \
+          \( \
+            -regextype posix-extended \
+            -regex '.*[.](bash|sh)' -and \
+            -type f \
+          \) \
+          -print | xargs shfmt -w
```

Comment:
The problem with this approach is it causes the loss of formatting for shell files without a file extension (which are relatively common). When passed the path to a folder, **shfmt** is able to identify these files from the [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)#Examples) in files while searching the folder recursively (a capability ShellCheck does not have).

I think the better approach will be to set these as ignore paths in the project's `.editorconfig` file instead. **shfmt** [is configured by that file](https://github.com/mvdan/sh/blob/v3.5.1/cmd/shfmt/shfmt.1.scd#:~:text=If%20any%20EditorConfig%20files%20are%20found), so it will not format any files that are under the ignored folders.

We must give consideration to any side effects setting these folders as ignored might have on other unrelated applications of the `.editorconfig` file (e.g., [the "Check General Formatting" template](https://github.com/arduino/tooling-project-assets/blob/main/workflow-templates/check-general-formatting-task.md)). However, I don't think ignoring these folders will cause any harm. I actually have adding them to the template `.editorconfig` file as an item on my "to do" list.